clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -946,8 +907,7 @@ public Map<TopicPartition, OffsetAndMetadata> committed(final Set<TopicPartition
                 timer);
```

Comment:
The timer is not used anywhere else. Maybe a deadline for this event would be simpler.

Reply:
Agreed. Using the deadline directly instead of a `Timer`.